contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -699,4 +767,158 @@ contract NodeOperatorsRegistry is INodeOperatorsRegistry, IsContract, AragonApp
         KEYS_OP_INDEX_POSITION.setStorageUint256(keysOpIndex + 1);
         emit KeysOpIndexSet(keysOpIndex + 1);
     }
+
+    function _trimUnusedNodeOperatorKeys(uint256 _nodeOperatorId) internal returns (uint64 trimmedKeys) {
+        uint64 totalSigningKeys = operators[_nodeOperatorId].totalSigningKeys;
+        uint64 usedSigningKeys = operators[_nodeOperatorId].usedSigningKeys;
+
+        // write only if update is needed
+        if (totalSigningKeys != usedSigningKeys) {
+            trimmedKeys = totalSigningKeys - usedSigningKeys;
+            operators[_nodeOperatorId].totalSigningKeys = usedSigningKeys; // discard unused keys
+            emit NodeOperatorTotalKeysTrimmed(_nodeOperatorId, trimmedKeys);
+        }
+    }
+
+    function _setTotalActiveKeys(uint64 _newTotalActiveKeys) internal {
+        keysUsageStats.totalActiveKeys = _newTotalActiveKeys;
+    }
+
+    function _updateTotalAvailableKeysCount() internal {
+        uint256 totalAvailableKeysCount;
+        uint256 activeNodeOperatorsCount = getNodeOperatorsCount();
+        for (uint256 i = 0; i < activeNodeOperatorsCount; ++i) {
+            totalAvailableKeysCount += _getNodeOperatorAvailableKeysCount(i);
+        }
+        keysUsageStats.totalAvailableKeys = to64(totalAvailableKeysCount);
+    }
+
+    function _getNodeOperatorAvailableKeysCount(uint256 _operatorId) internal view returns (uint256) {
+        NodeOperator storage operator = operators[_operatorId];
+        if (!operator.active) {
+            return 0;
+        }
+        uint256 operatorStakingLimit = operator.stakingLimit;
+        uint256 operatorUsedSigningKeys = operator.usedSigningKeys;
+        uint256 operatorTotalSigningKeys = operator.totalSigningKeys;
+        if (operatorUsedSigningKeys >= operatorStakingLimit || operatorUsedSigningKeys >= operatorTotalSigningKeys) {
+            return 0;
+        }
+
+        return
+            operatorStakingLimit < operatorTotalSigningKeys
```

Comment:
Maybe should explain why so, I didn't get it

Reply:
fixed